plugins/woocommerce/src/Admin/Features/OnboardingTasks/TaskList.php

```diff
@@ -370,6 +370,20 @@ public function get_keep_completed_task_list() {
 		return get_option( 'woocommerce_task_list_keep_completed', 'no' );
 	}
 
+	/**
+	 * Returns option to keep completed task list.
+	 *
+	 * @return string
+	 */
+	public function show_ces_feedback() {
+		if ( ! $this->is_complete() ) {
+			return false;
+		}
+		$allow_tracking = 'yes' === get_option( 'woocommerce_allow_tracking' );
+		$shown_ces_actions = get_option( 'woocommerce_ces_shown_for_actions', array() );
+		return $allow_tracking && ! in_array( 'store_setup', $shown_ces_actions, true );
```

Comment:
I think this might be UI-specific logic sneaking into the API.  The bit about `store_setup` is also not generic enough to be included in the `TaskList` builder class.

Reply:
You think so? isn't most of this TaskList class UI specific?
My thought was that this will be (most likely) the new standard for all the task lists, so it made the most sense to put it here so we don't specifically have to load the `woocommerce_ces_shown_for_actions` option on the front-end.
Now I do realize I am slightly contradicting myself here as we are updating the `woocommerce_ces_shown_for_actions` on the front-end, which could cause some confusion. I could create a separate API call for submitting task list setup (`store_setup`) as part of this and make use of that instead on the front-end.
That way we don't have logic floating around in two places.

Pushing this to the back-end was also to make the display look nicer, as I find on my local instance that the `get_option` api call can take a little while, so causing a bigger delay for the task list to show up.

Reply:
>isn't most of this TaskList class UI specific?

My thoughts are that it is not.  The task list provides data around what tasks are available.  It's up to the consumer in how this is displayed in the UI.

If the mobile app were to use this data, would the `woocommerce_ces_shown_for_actions` data make sense?  I would argue that it probably doesn't given the CES may or may not be displayed there.  This is a choice the client gets to make in terms of display.

>I find on my local instance that the get_option api call can take a little while

Good point.  I wonder if we can run some of these processes in parallel.  Another option is that if we did have a modal API we could opt to make this one of the necessary "preloaded" data items if it fits the criteria.  This will definitely be a hot topic for performance going forward.

Reply:
> If the mobile app were to use this data, would the woocommerce_ces_shown_for_actions data make sense? I would argue that it probably doesn't given the CES may or may not be displayed there. This is a choice the client gets to make in terms of display.

Good point, thanks for the example, I kinda forgot about the use case of the task list API being used outside of our current context.
In that sense the CES does sound a little to specific. handling it through the options for now would be fine with me.

> we could opt to make this one of the necessary "preloaded" data items if it fits the criteria.

I wouldn't mind adding it here, but I don't think it fits the criteria. Given it's only used on our home page.

Reply:
Ah I see.  In that case, would it be possible to forego passing `feedbackScore` back to the child and just rely on `submittedScore`?

On that same note `submittedScore` - could we rename this to `hasSubmitted` or just `isComplete`?  At first glance I assumed `submittedScore` was an actual score.

Reply:
@joshuatf I assume you meant to reply this in a different thread?

Reply:
But to continue that convo:
> Ah I see. In that case, would it be possible to forego passing feedbackScore back to the child and just rely on submittedScore?

Yeah, instead of actually keeping a local state in the simple one, we could just only do the callback and handle the submission in the parent.

Reply:
As per our discussion this logic has been removed.